src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionViewModel.cs

```diff
@@ -193,20 +218,37 @@ internal sealed class ConnectionViewModel : BindableBase, IConnectionViewModel {
                 || !Path.EqualsIgnoreCase(_connection?.Path)
                 || !RCommandLineArguments.EqualsIgnoreCase(_connection?.RCommandLineArguments);
 
+            IsNameValid = IsValidConnectionName(Name);
+            if (!IsNameValid) {
+                NameTextBoxTooltip = string.IsNullOrEmpty(Name)
+                    ? Resources.ConnectionManager_ShouldHaveName
+                    : Resources.ConnectionManager_InvalidName;
+            } else {
+                NameTextBoxTooltip = null;
+            }
+
             Uri uri;
-            var isPathValid = Uri.TryCreate(Path, UriKind.Absolute, out uri);
-            if (!IsValidConnectionName(Name)) {
-                IsValid = false;
-                SaveButtonTooltip = Resources.ConnectionManager_ShouldHaveName;
-            } else if (!isPathValid) {
-                IsValid = false;
-                SaveButtonTooltip = Resources.ConnectionManager_ShouldHavePath;
+            IsPathValid = IsValidConnectionUrl(Path, out uri);
```

Comment:
I think `Uri.TryCreate` is enough here, cause `UpdateCalculated` will be called after we have tried to fix the path.